py-polars/src/npy.rs

```diff
@@ -14,12 +15,12 @@ use std::{mem, ptr};
 /// All elements in the array are non initialized
 ///
 /// The array is also writable from Python.
-pub unsafe fn aligned_array<T: Element>(py: Python<'_>, size: usize) -> (&PyArray1<T>, *mut T) {
-    let t_size = std::mem::size_of::<T>();
-    let capacity = size * t_size;
-    let ptr = alloc::allocate_aligned::<u8>(capacity).as_ptr() as *mut T;
-    let mut buf = Vec::from_raw_parts(ptr, 0, capacity);
```

Comment:
I also removed this bit, as this is usually UB, and we do not really need the `Vec`: we can pass the pointer to `PyArray_New` directly, thereby removing this indirection.